koa-crud/src/use-cases/promos/select-one-promo.ts

```diff
@@ -1,13 +1,24 @@
-import { UseCase, PromoDocument, PromosStore } from '../../types';
+import { PromoDocument } from '../../lib/mongoose/models/promo';
+import { PromoStore } from '../../data-access/mongoose/promos/actions';
 import { PromoNotFoundError } from '../../custom-errors';
 
+type Input = {
+  id: string;
+  info?;
+  source?;
+};
+
+type Output = PromoDocument;
+
+export type SelectOnePromoUseCase = (input: Input) => Promise<Output>;
```

Comment:
Since most of the usecase type is almost the same why not add a generic type for UseCase like this:
```ts
type UseCase<TInput, TOutput> = (input: TInput) => Promise<TOutput>;

export type SelectOnePromoUseCase = UseCase<Input, Output>;
```

Reply:
noted on this